// pipedrive/dist/model/AdditionalDataWithOffsetPagination.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AdditionalData = _interopRequireDefault(require("./AdditionalData"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The AdditionalDataWithOffsetPagination model module.
 * @module model/AdditionalDataWithOffsetPagination
 * @version 1.0.0
 */
var AdditionalDataWithOffsetPagination = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AdditionalDataWithOffsetPagination</code>.
   * @alias module:model/AdditionalDataWithOffsetPagination
   */
  function AdditionalDataWithOffsetPagination() {
    (0, _classCallCheck2["default"])(this, AdditionalDataWithOffsetPagination);
    AdditionalDataWithOffsetPagination.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(AdditionalDataWithOffsetPagination, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AdditionalDataWithOffsetPagination</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AdditionalDataWithOffsetPagination} obj Optional instance to populate.
     * @return {module:model/AdditionalDataWithOffsetPagination} The populated <code>AdditionalDataWithOffsetPagination</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AdditionalDataWithOffsetPagination();
        if (data.hasOwnProperty('pagination')) {
          obj['pagination'] = _ApiClient["default"].convertToType(data['pagination'], _AdditionalData["default"]);
          delete data['pagination'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return AdditionalDataWithOffsetPagination;
}();
/**
 * The pagination details of the list
 * @member {module:model/AdditionalData} pagination
 */
AdditionalDataWithOffsetPagination.prototype['pagination'] = undefined;
var _default = AdditionalDataWithOffsetPagination;
exports["default"] = _default;
