// pipedrive/dist/model/AddRole.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The AddRole model module.
 * @module model/AddRole
 * @version 1.0.0
 */
var AddRole = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AddRole</code>.
   * The details of the role
   * @alias module:model/AddRole
   * @param name {String} The name of the role
   */
  function AddRole(name) {
    (0, _classCallCheck2["default"])(this, AddRole);
    AddRole.initialize(this, name);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(AddRole, null, [{
    key: "initialize",
    value: function initialize(obj, name) {
      obj['name'] = name;
    }

    /**
     * Constructs a <code>AddRole</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AddRole} obj Optional instance to populate.
     * @return {module:model/AddRole} The populated <code>AddRole</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AddRole();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }
        if (data.hasOwnProperty('parent_role_id')) {
          obj['parent_role_id'] = _ApiClient["default"].convertToType(data['parent_role_id'], 'Number');
          delete data['parent_role_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return AddRole;
}();
/**
 * The name of the role
 * @member {String} name
 */
AddRole.prototype['name'] = undefined;

/**
 * The ID of the parent role
 * @member {Number} parent_role_id
 */
AddRole.prototype['parent_role_id'] = undefined;
var _default = AddRole;
exports["default"] = _default;
