// pipedrive/dist/model/CallLogsResponse.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _CallLogsResponseAdditionalData = _interopRequireDefault(require("./CallLogsResponseAdditionalData"));
var _ResponseCallLogObject = _interopRequireDefault(require("./ResponseCallLogObject"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The CallLogsResponse model module.
 * @module model/CallLogsResponse
 * @version 1.0.0
 */
var CallLogsResponse = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>CallLogsResponse</code>.
   * @alias module:model/CallLogsResponse
   */
  function CallLogsResponse() {
    (0, _classCallCheck2["default"])(this, CallLogsResponse);
    CallLogsResponse.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(CallLogsResponse, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>CallLogsResponse</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/CallLogsResponse} obj Optional instance to populate.
     * @return {module:model/CallLogsResponse} The populated <code>CallLogsResponse</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new CallLogsResponse();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_ResponseCallLogObject["default"]]);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _CallLogsResponseAdditionalData["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return CallLogsResponse;
}();
/**
 * If the response is successful or not
 * @member {Boolean} success
 */
CallLogsResponse.prototype['success'] = undefined;

/**
 * @member {Array.<module:model/ResponseCallLogObject>} data
 */
CallLogsResponse.prototype['data'] = undefined;

/**
 * @member {module:model/CallLogsResponseAdditionalData} additional_data
 */
CallLogsResponse.prototype['additional_data'] = undefined;
var _default = CallLogsResponse;
exports["default"] = _default;
