// pipedrive/dist/model/DeleteRoleAssignmentRequest.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The DeleteRoleAssignmentRequest model module.
 * @module model/DeleteRoleAssignmentRequest
 * @version 1.0.0
 */
var DeleteRoleAssignmentRequest = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>DeleteRoleAssignmentRequest</code>.
   * @alias module:model/DeleteRoleAssignmentRequest
   * @param userId {Number} The ID of the user
   */
  function DeleteRoleAssignmentRequest(userId) {
    (0, _classCallCheck2["default"])(this, DeleteRoleAssignmentRequest);
    DeleteRoleAssignmentRequest.initialize(this, userId);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(DeleteRoleAssignmentRequest, null, [{
    key: "initialize",
    value: function initialize(obj, userId) {
      obj['user_id'] = userId;
    }

    /**
     * Constructs a <code>DeleteRoleAssignmentRequest</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/DeleteRoleAssignmentRequest} obj Optional instance to populate.
     * @return {module:model/DeleteRoleAssignmentRequest} The populated <code>DeleteRoleAssignmentRequest</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new DeleteRoleAssignmentRequest();
        if (data.hasOwnProperty('user_id')) {
          obj['user_id'] = _ApiClient["default"].convertToType(data['user_id'], 'Number');
          delete data['user_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return DeleteRoleAssignmentRequest;
}();
/**
 * The ID of the user
 * @member {Number} user_id
 */
DeleteRoleAssignmentRequest.prototype['user_id'] = undefined;
var _default = DeleteRoleAssignmentRequest;
exports["default"] = _default;
