// pipedrive/dist/model/GetRolePipelinesAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The GetRolePipelinesAllOfData model module.
 * @module model/GetRolePipelinesAllOfData
 * @version 1.0.0
 */
var GetRolePipelinesAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>GetRolePipelinesAllOfData</code>.
   * The response data
   * @alias module:model/GetRolePipelinesAllOfData
   */
  function GetRolePipelinesAllOfData() {
    (0, _classCallCheck2["default"])(this, GetRolePipelinesAllOfData);
    GetRolePipelinesAllOfData.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(GetRolePipelinesAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>GetRolePipelinesAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/GetRolePipelinesAllOfData} obj Optional instance to populate.
     * @return {module:model/GetRolePipelinesAllOfData} The populated <code>GetRolePipelinesAllOfData</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new GetRolePipelinesAllOfData();
        if (data.hasOwnProperty('pipeline_ids')) {
          obj['pipeline_ids'] = _ApiClient["default"].convertToType(data['pipeline_ids'], ['Number']);
          delete data['pipeline_ids'];
        }
        if (data.hasOwnProperty('visible')) {
          obj['visible'] = _ApiClient["default"].convertToType(data['visible'], 'Boolean');
          delete data['visible'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return GetRolePipelinesAllOfData;
}();
/**
 * Either visible or hidden pipeline ids
 * @member {Array.<Number>} pipeline_ids
 */
GetRolePipelinesAllOfData.prototype['pipeline_ids'] = undefined;

/**
 * Whether visible or hidden pipeline ids were returned
 * @member {Boolean} visible
 */
GetRolePipelinesAllOfData.prototype['visible'] = undefined;
var _default = GetRolePipelinesAllOfData;
exports["default"] = _default;
