// pipedrive/dist/model/InlineResponse2002.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _AdditionalDataWithCursorPagination = _interopRequireDefault(require("./AdditionalDataWithCursorPagination"));
var _PersonsCollectionResponseObject = _interopRequireDefault(require("./PersonsCollectionResponseObject"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The InlineResponse2002 model module.
 * @module model/InlineResponse2002
 * @version 1.0.0
 */
var InlineResponse2002 = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>InlineResponse2002</code>.
   * @alias module:model/InlineResponse2002
   */
  function InlineResponse2002() {
    (0, _classCallCheck2["default"])(this, InlineResponse2002);
    InlineResponse2002.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(InlineResponse2002, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>InlineResponse2002</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/InlineResponse2002} obj Optional instance to populate.
     * @return {module:model/InlineResponse2002} The populated <code>InlineResponse2002</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new InlineResponse2002();
        if (data.hasOwnProperty('success')) {
          obj['success'] = _ApiClient["default"].convertToType(data['success'], 'Boolean');
          delete data['success'];
        }
        if (data.hasOwnProperty('data')) {
          obj['data'] = _ApiClient["default"].convertToType(data['data'], [_PersonsCollectionResponseObject["default"]]);
          delete data['data'];
        }
        if (data.hasOwnProperty('additional_data')) {
          obj['additional_data'] = _AdditionalDataWithCursorPagination["default"].constructFromObject(data['additional_data']);
          delete data['additional_data'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return InlineResponse2002;
}();
/**
 * @member {Boolean} success
 */
InlineResponse2002.prototype['success'] = undefined;

/**
 * @member {Array.<module:model/PersonsCollectionResponseObject>} data
 */
InlineResponse2002.prototype['data'] = undefined;

/**
 * @member {module:model/AdditionalDataWithCursorPagination} additional_data
 */
InlineResponse2002.prototype['additional_data'] = undefined;
var _default = InlineResponse2002;
exports["default"] = _default;
