// pipedrive/dist/model/MailThreadDeleteAllOfData.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The MailThreadDeleteAllOfData model module.
 * @module model/MailThreadDeleteAllOfData
 * @version 1.0.0
 */
var MailThreadDeleteAllOfData = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>MailThreadDeleteAllOfData</code>.
   * @alias module:model/MailThreadDeleteAllOfData
   */
  function MailThreadDeleteAllOfData() {
    (0, _classCallCheck2["default"])(this, MailThreadDeleteAllOfData);
    MailThreadDeleteAllOfData.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(MailThreadDeleteAllOfData, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>MailThreadDeleteAllOfData</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/MailThreadDeleteAllOfData} obj Optional instance to populate.
     * @return {module:model/MailThreadDeleteAllOfData} The populated <code>MailThreadDeleteAllOfData</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new MailThreadDeleteAllOfData();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return MailThreadDeleteAllOfData;
}();
/**
 * The ID of the deleted mail thread
 * @member {Number} id
 */
MailThreadDeleteAllOfData.prototype['id'] = undefined;
var _default = MailThreadDeleteAllOfData;
exports["default"] = _default;
