// pipedrive/dist/model/OrganizationsCollectionResponseObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _OrganizationAddressInfo = _interopRequireDefault(require("./OrganizationAddressInfo"));
var _OrganizationsCollectionResponseObjectAllOf = _interopRequireDefault(require("./OrganizationsCollectionResponseObjectAllOf"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The OrganizationsCollectionResponseObject model module.
 * @module model/OrganizationsCollectionResponseObject
 * @version 1.0.0
 */
var OrganizationsCollectionResponseObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationsCollectionResponseObject</code>.
   * @alias module:model/OrganizationsCollectionResponseObject
   * @implements module:model/OrganizationAddressInfo
   * @implements module:model/OrganizationsCollectionResponseObjectAllOf
   */
  function OrganizationsCollectionResponseObject() {
    (0, _classCallCheck2["default"])(this, OrganizationsCollectionResponseObject);
    _OrganizationAddressInfo["default"].initialize(this);
    _OrganizationsCollectionResponseObjectAllOf["default"].initialize(this);
    OrganizationsCollectionResponseObject.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(OrganizationsCollectionResponseObject, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>OrganizationsCollectionResponseObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationsCollectionResponseObject} obj Optional instance to populate.
     * @return {module:model/OrganizationsCollectionResponseObject} The populated <code>OrganizationsCollectionResponseObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationsCollectionResponseObject();
        _OrganizationAddressInfo["default"].constructFromObject(data, obj);
        _OrganizationsCollectionResponseObjectAllOf["default"].constructFromObject(data, obj);
        if (data.hasOwnProperty('address')) {
          obj['address'] = _ApiClient["default"].convertToType(data['address'], 'String');
          delete data['address'];
        }
        if (data.hasOwnProperty('address_subpremise')) {
          obj['address_subpremise'] = _ApiClient["default"].convertToType(data['address_subpremise'], 'String');
          delete data['address_subpremise'];
        }
        if (data.hasOwnProperty('address_street_number')) {
          obj['address_street_number'] = _ApiClient["default"].convertToType(data['address_street_number'], 'String');
          delete data['address_street_number'];
        }
        if (data.hasOwnProperty('address_route')) {
          obj['address_route'] = _ApiClient["default"].convertToType(data['address_route'], 'String');
          delete data['address_route'];
        }
        if (data.hasOwnProperty('address_sublocality')) {
          obj['address_sublocality'] = _ApiClient["default"].convertToType(data['address_sublocality'], 'String');
          delete data['address_sublocality'];
        }
        if (data.hasOwnProperty('address_locality')) {
          obj['address_locality'] = _ApiClient["default"].convertToType(data['address_locality'], 'String');
          delete data['address_locality'];
        }
        if (data.hasOwnProperty('address_admin_area_level_1')) {
          obj['address_admin_area_level_1'] = _ApiClient["default"].convertToType(data['address_admin_area_level_1'], 'String');
          delete data['address_admin_area_level_1'];
        }
        if (data.hasOwnProperty('address_admin_area_level_2')) {
          obj['address_admin_area_level_2'] = _ApiClient["default"].convertToType(data['address_admin_area_level_2'], 'String');
          delete data['address_admin_area_level_2'];
        }
        if (data.hasOwnProperty('address_country')) {
          obj['address_country'] = _ApiClient["default"].convertToType(data['address_country'], 'String');
          delete data['address_country'];
        }
        if (data.hasOwnProperty('address_postal_code')) {
          obj['address_postal_code'] = _ApiClient["default"].convertToType(data['address_postal_code'], 'String');
          delete data['address_postal_code'];
        }
        if (data.hasOwnProperty('address_formatted_address')) {
          obj['address_formatted_address'] = _ApiClient["default"].convertToType(data['address_formatted_address'], 'String');
          delete data['address_formatted_address'];
        }
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }
        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }
        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }
        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }
        if (data.hasOwnProperty('delete_time')) {
          obj['delete_time'] = _ApiClient["default"].convertToType(data['delete_time'], 'String');
          delete data['delete_time'];
        }
        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }
        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'String');
          delete data['visible_to'];
        }
        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'Number');
          delete data['label'];
        }
        if (data.hasOwnProperty('cc_email')) {
          obj['cc_email'] = _ApiClient["default"].convertToType(data['cc_email'], 'String');
          delete data['cc_email'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return OrganizationsCollectionResponseObject;
}();
/**
 * The full address of the organization
 * @member {String} address
 */
OrganizationsCollectionResponseObject.prototype['address'] = undefined;

/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */
OrganizationsCollectionResponseObject.prototype['address_subpremise'] = undefined;

/**
 * The street number of the organization location
 * @member {String} address_street_number
 */
OrganizationsCollectionResponseObject.prototype['address_street_number'] = undefined;

/**
 * The route of the organization location
 * @member {String} address_route
 */
OrganizationsCollectionResponseObject.prototype['address_route'] = undefined;

/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */
OrganizationsCollectionResponseObject.prototype['address_sublocality'] = undefined;

/**
 * The locality of the organization location
 * @member {String} address_locality
 */
OrganizationsCollectionResponseObject.prototype['address_locality'] = undefined;

/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */
OrganizationsCollectionResponseObject.prototype['address_admin_area_level_1'] = undefined;

/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */
OrganizationsCollectionResponseObject.prototype['address_admin_area_level_2'] = undefined;

/**
 * The country of the organization location
 * @member {String} address_country
 */
OrganizationsCollectionResponseObject.prototype['address_country'] = undefined;

/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */
OrganizationsCollectionResponseObject.prototype['address_postal_code'] = undefined;

/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */
OrganizationsCollectionResponseObject.prototype['address_formatted_address'] = undefined;

/**
 * The ID of the organization
 * @member {Number} id
 */
OrganizationsCollectionResponseObject.prototype['id'] = undefined;

/**
 * Whether the organization is active or not
 * @member {Boolean} active_flag
 */
OrganizationsCollectionResponseObject.prototype['active_flag'] = undefined;

/**
 * The ID of the owner
 * @member {Number} owner_id
 */
OrganizationsCollectionResponseObject.prototype['owner_id'] = undefined;

/**
 * The name of the organization
 * @member {String} name
 */
OrganizationsCollectionResponseObject.prototype['name'] = undefined;

/**
 * The last updated date and time of the organization. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} update_time
 */
OrganizationsCollectionResponseObject.prototype['update_time'] = undefined;

/**
 * The date and time this organization was deleted. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} delete_time
 */
OrganizationsCollectionResponseObject.prototype['delete_time'] = undefined;

/**
 * The date and time when the organization was added/created. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */
OrganizationsCollectionResponseObject.prototype['add_time'] = undefined;

/**
 * The visibility group ID of who can see the organization
 * @member {String} visible_to
 */
OrganizationsCollectionResponseObject.prototype['visible_to'] = undefined;

/**
 * The label assigned to the organization
 * @member {Number} label
 */
OrganizationsCollectionResponseObject.prototype['label'] = undefined;

/**
 * The BCC email associated with the organization
 * @member {String} cc_email
 */
OrganizationsCollectionResponseObject.prototype['cc_email'] = undefined;

// Implement OrganizationAddressInfo interface:
/**
 * The full address of the organization
 * @member {String} address
 */
_OrganizationAddressInfo["default"].prototype['address'] = undefined;
/**
 * The sub-premise of the organization location
 * @member {String} address_subpremise
 */
_OrganizationAddressInfo["default"].prototype['address_subpremise'] = undefined;
/**
 * The street number of the organization location
 * @member {String} address_street_number
 */
_OrganizationAddressInfo["default"].prototype['address_street_number'] = undefined;
/**
 * The route of the organization location
 * @member {String} address_route
 */
_OrganizationAddressInfo["default"].prototype['address_route'] = undefined;
/**
 * The sub-locality of the organization location
 * @member {String} address_sublocality
 */
_OrganizationAddressInfo["default"].prototype['address_sublocality'] = undefined;
/**
 * The locality of the organization location
 * @member {String} address_locality
 */
_OrganizationAddressInfo["default"].prototype['address_locality'] = undefined;
/**
 * The level 1 admin area of the organization location
 * @member {String} address_admin_area_level_1
 */
_OrganizationAddressInfo["default"].prototype['address_admin_area_level_1'] = undefined;
/**
 * The level 2 admin area of the organization location
 * @member {String} address_admin_area_level_2
 */
_OrganizationAddressInfo["default"].prototype['address_admin_area_level_2'] = undefined;
/**
 * The country of the organization location
 * @member {String} address_country
 */
_OrganizationAddressInfo["default"].prototype['address_country'] = undefined;
/**
 * The postal code of the organization location
 * @member {String} address_postal_code
 */
_OrganizationAddressInfo["default"].prototype['address_postal_code'] = undefined;
/**
 * The formatted organization location
 * @member {String} address_formatted_address
 */
_OrganizationAddressInfo["default"].prototype['address_formatted_address'] = undefined;
// Implement OrganizationsCollectionResponseObjectAllOf interface:
/**
 * The ID of the organization
 * @member {Number} id
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['id'] = undefined;
/**
 * Whether the organization is active or not
 * @member {Boolean} active_flag
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['active_flag'] = undefined;
/**
 * The ID of the owner
 * @member {Number} owner_id
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['owner_id'] = undefined;
/**
 * The name of the organization
 * @member {String} name
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['name'] = undefined;
/**
 * The last updated date and time of the organization. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} update_time
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['update_time'] = undefined;
/**
 * The date and time this organization was deleted. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} delete_time
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['delete_time'] = undefined;
/**
 * The date and time when the organization was added/created. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['add_time'] = undefined;
/**
 * The visibility group ID of who can see the organization
 * @member {String} visible_to
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['visible_to'] = undefined;
/**
 * The label assigned to the organization
 * @member {Number} label
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['label'] = undefined;
/**
 * The BCC email associated with the organization
 * @member {String} cc_email
 */
_OrganizationsCollectionResponseObjectAllOf["default"].prototype['cc_email'] = undefined;
var _default = OrganizationsCollectionResponseObject;
exports["default"] = _default;
