// pipedrive/dist/model/OrganizationsCollectionResponseObjectAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The OrganizationsCollectionResponseObjectAllOf model module.
 * @module model/OrganizationsCollectionResponseObjectAllOf
 * @version 1.0.0
 */
var OrganizationsCollectionResponseObjectAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>OrganizationsCollectionResponseObjectAllOf</code>.
   * @alias module:model/OrganizationsCollectionResponseObjectAllOf
   */
  function OrganizationsCollectionResponseObjectAllOf() {
    (0, _classCallCheck2["default"])(this, OrganizationsCollectionResponseObjectAllOf);
    OrganizationsCollectionResponseObjectAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(OrganizationsCollectionResponseObjectAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>OrganizationsCollectionResponseObjectAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/OrganizationsCollectionResponseObjectAllOf} obj Optional instance to populate.
     * @return {module:model/OrganizationsCollectionResponseObjectAllOf} The populated <code>OrganizationsCollectionResponseObjectAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new OrganizationsCollectionResponseObjectAllOf();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }
        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }
        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }
        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }
        if (data.hasOwnProperty('delete_time')) {
          obj['delete_time'] = _ApiClient["default"].convertToType(data['delete_time'], 'String');
          delete data['delete_time'];
        }
        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }
        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'String');
          delete data['visible_to'];
        }
        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'Number');
          delete data['label'];
        }
        if (data.hasOwnProperty('cc_email')) {
          obj['cc_email'] = _ApiClient["default"].convertToType(data['cc_email'], 'String');
          delete data['cc_email'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return OrganizationsCollectionResponseObjectAllOf;
}();
/**
 * The ID of the organization
 * @member {Number} id
 */
OrganizationsCollectionResponseObjectAllOf.prototype['id'] = undefined;

/**
 * Whether the organization is active or not
 * @member {Boolean} active_flag
 */
OrganizationsCollectionResponseObjectAllOf.prototype['active_flag'] = undefined;

/**
 * The ID of the owner
 * @member {Number} owner_id
 */
OrganizationsCollectionResponseObjectAllOf.prototype['owner_id'] = undefined;

/**
 * The name of the organization
 * @member {String} name
 */
OrganizationsCollectionResponseObjectAllOf.prototype['name'] = undefined;

/**
 * The last updated date and time of the organization. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} update_time
 */
OrganizationsCollectionResponseObjectAllOf.prototype['update_time'] = undefined;

/**
 * The date and time this organization was deleted. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} delete_time
 */
OrganizationsCollectionResponseObjectAllOf.prototype['delete_time'] = undefined;

/**
 * The date and time when the organization was added/created. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */
OrganizationsCollectionResponseObjectAllOf.prototype['add_time'] = undefined;

/**
 * The visibility group ID of who can see the organization
 * @member {String} visible_to
 */
OrganizationsCollectionResponseObjectAllOf.prototype['visible_to'] = undefined;

/**
 * The label assigned to the organization
 * @member {Number} label
 */
OrganizationsCollectionResponseObjectAllOf.prototype['label'] = undefined;

/**
 * The BCC email associated with the organization
 * @member {String} cc_email
 */
OrganizationsCollectionResponseObjectAllOf.prototype['cc_email'] = undefined;
var _default = OrganizationsCollectionResponseObjectAllOf;
exports["default"] = _default;
