// pipedrive/dist/model/PersonsCollectionResponseObject.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
var _BasePersonItemPhone = _interopRequireDefault(require("./BasePersonItemPhone"));
var _BasicPersonEmail = _interopRequireDefault(require("./BasicPersonEmail"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The PersonsCollectionResponseObject model module.
 * @module model/PersonsCollectionResponseObject
 * @version 1.0.0
 */
var PersonsCollectionResponseObject = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>PersonsCollectionResponseObject</code>.
   * @alias module:model/PersonsCollectionResponseObject
   */
  function PersonsCollectionResponseObject() {
    (0, _classCallCheck2["default"])(this, PersonsCollectionResponseObject);
    PersonsCollectionResponseObject.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(PersonsCollectionResponseObject, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>PersonsCollectionResponseObject</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/PersonsCollectionResponseObject} obj Optional instance to populate.
     * @return {module:model/PersonsCollectionResponseObject} The populated <code>PersonsCollectionResponseObject</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new PersonsCollectionResponseObject();
        if (data.hasOwnProperty('id')) {
          obj['id'] = _ApiClient["default"].convertToType(data['id'], 'Number');
          delete data['id'];
        }
        if (data.hasOwnProperty('active_flag')) {
          obj['active_flag'] = _ApiClient["default"].convertToType(data['active_flag'], 'Boolean');
          delete data['active_flag'];
        }
        if (data.hasOwnProperty('owner_id')) {
          obj['owner_id'] = _ApiClient["default"].convertToType(data['owner_id'], 'Number');
          delete data['owner_id'];
        }
        if (data.hasOwnProperty('org_id')) {
          obj['org_id'] = _ApiClient["default"].convertToType(data['org_id'], 'Number');
          delete data['org_id'];
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }
        if (data.hasOwnProperty('email')) {
          obj['email'] = _ApiClient["default"].convertToType(data['email'], [_BasicPersonEmail["default"]]);
          delete data['email'];
        }
        if (data.hasOwnProperty('phone')) {
          obj['phone'] = _ApiClient["default"].convertToType(data['phone'], [_BasePersonItemPhone["default"]]);
          delete data['phone'];
        }
        if (data.hasOwnProperty('update_time')) {
          obj['update_time'] = _ApiClient["default"].convertToType(data['update_time'], 'String');
          delete data['update_time'];
        }
        if (data.hasOwnProperty('delete_time')) {
          obj['delete_time'] = _ApiClient["default"].convertToType(data['delete_time'], 'String');
          delete data['delete_time'];
        }
        if (data.hasOwnProperty('add_time')) {
          obj['add_time'] = _ApiClient["default"].convertToType(data['add_time'], 'String');
          delete data['add_time'];
        }
        if (data.hasOwnProperty('visible_to')) {
          obj['visible_to'] = _ApiClient["default"].convertToType(data['visible_to'], 'String');
          delete data['visible_to'];
        }
        if (data.hasOwnProperty('picture_id')) {
          obj['picture_id'] = _ApiClient["default"].convertToType(data['picture_id'], 'Number');
          delete data['picture_id'];
        }
        if (data.hasOwnProperty('label')) {
          obj['label'] = _ApiClient["default"].convertToType(data['label'], 'Number');
          delete data['label'];
        }
        if (data.hasOwnProperty('cc_email')) {
          obj['cc_email'] = _ApiClient["default"].convertToType(data['cc_email'], 'String');
          delete data['cc_email'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return PersonsCollectionResponseObject;
}();
/**
 * The ID of the person
 * @member {Number} id
 */
PersonsCollectionResponseObject.prototype['id'] = undefined;

/**
 * Whether the person is active or not
 * @member {Boolean} active_flag
 */
PersonsCollectionResponseObject.prototype['active_flag'] = undefined;

/**
 * The ID of the owner related to the person
 * @member {Number} owner_id
 */
PersonsCollectionResponseObject.prototype['owner_id'] = undefined;

/**
 * The ID of the organization related to the person
 * @member {Number} org_id
 */
PersonsCollectionResponseObject.prototype['org_id'] = undefined;

/**
 * The name of the person
 * @member {String} name
 */
PersonsCollectionResponseObject.prototype['name'] = undefined;

/**
 * An email address as a string or an array of email objects related to the person. The structure of the array is as follows: `[{ \"value\": \"mail@example.com\", \"primary\": \"true\", \"label\": \"main\" }]`. Please note that only `value` is required.
 * @member {Array.<module:model/BasicPersonEmail>} email
 */
PersonsCollectionResponseObject.prototype['email'] = undefined;

/**
 * A phone number supplied as a string or an array of phone objects related to the person. The structure of the array is as follows: `[{ \"value\": \"12345\", \"primary\": \"true\", \"label\": \"mobile\" }]`. Please note that only `value` is required.
 * @member {Array.<module:model/BasePersonItemPhone>} phone
 */
PersonsCollectionResponseObject.prototype['phone'] = undefined;

/**
 * The last updated date and time of the person. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} update_time
 */
PersonsCollectionResponseObject.prototype['update_time'] = undefined;

/**
 * The date and time this person was deleted. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} delete_time
 */
PersonsCollectionResponseObject.prototype['delete_time'] = undefined;

/**
 * The date and time when the person was added/created. Format: YYYY-MM-DD HH:MM:SS
 * @member {String} add_time
 */
PersonsCollectionResponseObject.prototype['add_time'] = undefined;

/**
 * The visibility group ID of who can see the person
 * @member {String} visible_to
 */
PersonsCollectionResponseObject.prototype['visible_to'] = undefined;

/**
 * The ID of the picture associated with the item
 * @member {Number} picture_id
 */
PersonsCollectionResponseObject.prototype['picture_id'] = undefined;

/**
 * The label assigned to the person
 * @member {Number} label
 */
PersonsCollectionResponseObject.prototype['label'] = undefined;

/**
 * The BCC email associated with the person
 * @member {String} cc_email
 */
PersonsCollectionResponseObject.prototype['cc_email'] = undefined;
var _default = PersonsCollectionResponseObject;
exports["default"] = _default;
