// pipedrive/dist/model/UpdateOrganizationAllOf.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UpdateOrganizationAllOf model module.
 * @module model/UpdateOrganizationAllOf
 * @version 1.0.0
 */
var UpdateOrganizationAllOf = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UpdateOrganizationAllOf</code>.
   * @alias module:model/UpdateOrganizationAllOf
   */
  function UpdateOrganizationAllOf() {
    (0, _classCallCheck2["default"])(this, UpdateOrganizationAllOf);
    UpdateOrganizationAllOf.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UpdateOrganizationAllOf, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UpdateOrganizationAllOf</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UpdateOrganizationAllOf} obj Optional instance to populate.
     * @return {module:model/UpdateOrganizationAllOf} The populated <code>UpdateOrganizationAllOf</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UpdateOrganizationAllOf();
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
          delete data['name'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UpdateOrganizationAllOf;
}();
/**
 * The name of the organization
 * @member {String} name
 */
UpdateOrganizationAllOf.prototype['name'] = undefined;
var _default = UpdateOrganizationAllOf;
exports["default"] = _default;
