// pipedrive/dist/model/UserAccess.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));
var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
/**
 * Pipedrive API v1
 * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
 *
 * The version of the OpenAPI document: 1.0.0
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 *
 */
/**
 * The UserAccess model module.
 * @module model/UserAccess
 * @version 1.0.0
 */
var UserAccess = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>UserAccess</code>.
   * @alias module:model/UserAccess
   */
  function UserAccess() {
    (0, _classCallCheck2["default"])(this, UserAccess);
    UserAccess.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  (0, _createClass2["default"])(UserAccess, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>UserAccess</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/UserAccess} obj Optional instance to populate.
     * @return {module:model/UserAccess} The populated <code>UserAccess</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new UserAccess();
        if (data.hasOwnProperty('app')) {
          obj['app'] = _ApiClient["default"].convertToType(data['app'], 'String');
          delete data['app'];
        }
        if (data.hasOwnProperty('admin')) {
          obj['admin'] = _ApiClient["default"].convertToType(data['admin'], 'Boolean');
          delete data['admin'];
        }
        if (data.hasOwnProperty('permission_set_id')) {
          obj['permission_set_id'] = _ApiClient["default"].convertToType(data['permission_set_id'], 'String');
          delete data['permission_set_id'];
        }
        if (Object.keys(data).length > 0) {
          Object.assign(obj, data);
        }
      }
      return obj;
    }
  }]);
  return UserAccess;
}();
/**
 * @member {module:model/UserAccess.AppEnum} app
 */
UserAccess.prototype['app'] = undefined;

/**
 * @member {Boolean} admin
 */
UserAccess.prototype['admin'] = undefined;

/**
 * @member {String} permission_set_id
 */
UserAccess.prototype['permission_set_id'] = undefined;

/**
 * Allowed values for the <code>app</code> property.
 * @enum {String}
 * @readonly
 */
UserAccess['AppEnum'] = {
  /**
   * value: "sales"
   * @const
   */
  "sales": "sales",
  /**
   * value: "projects"
   * @const
   */
  "projects": "projects",
  /**
   * value: "campaigns"
   * @const
   */
  "campaigns": "campaigns",
  /**
   * value: "global"
   * @const
   */
  "global": "global",
  /**
   * value: "account_settings"
   * @const
   */
  "account_settings": "account_settings"
};
var _default = UserAccess;
exports["default"] = _default;
